R/db2.R

```diff
@@ -123,21 +123,21 @@ report_db2_init <- function(con, config, must_create = FALSE, validate = TRUE) {
 }
 
 
-report_db2_init_create <- function(con, config) {
-  if (config$destination$driver[[1]] == "RSQLite") {
-    dialect <- "sqlite"
-  } else {
-    dialect <- "postgres"
-  }
+report_db2_init_create <- function(con, config, dialect) {
   dat <- orderly_schema_prepare(config$fields, dialect)
   dat$values$changelog_label <- config$changelog
 
+  DBI::dbBegin(con)
```

Comment:
Now tested - failure in initialisation left things in an unrecoverable state. Using a transaction now to stop that